plugins/modules/postgresql_alter_system.py

```diff
@@ -572,7 +572,16 @@ def convert_ret_vals(attrs):
         if str_contains_float(attrs[elem]):
             attrs[elem] = float(attrs[elem])
         else:
-            attrs[elem] = int(attrs[elem])
+            try:
+                attrs[elem] = int(attrs[elem])
+            except ValueError:
```

Comment:
If the cast to `int()` fails, it is likely or at least possible that PostgreSQL doesn't like the value either.
It doesn't help if the Ansible code passes, just to fail moments later in the database.

Reply:
@andreasscherbaum shouldn't we just return what DB returns, because if people run a select against max_val it'll return "1.79769e+308". I suggest leaving it to PG devs and just return what DB returns in similar cases. Thoughts?

Reply:
You have a point there, and this is just the return value. If someone uses this same value to set a parameter, this will work. If someone uses an arbitrary value, this might fail - but that is a different problem.